### DISCUSSION TITLE
nextjs 13 prerender error while deploying on vercel

### DISCUSSION THREAD

**Discussion by arooj-12** [vercel/next.js#58988] [Help]
### Summary

i didnt have any error while on localhost. but when i start deploying my frontend on vercel i got prerender error on app/(routes)/cart/page: /cart
/_not-found
on vs code i have cart folder which consists of components folder and page.tsx file. can someone please help i have been stuck on this since 2 days.

### Additional information

```js
https://nextjs.org/docs/messages/prerender-error
this nextjs doc states ways to fix this error but i dont have "pages" folder in my project and the last 2 options as well i am not using it
```

**Comment by icyJoseph** [comment]
The logs are a still pending to update I guess, but could you share more of the errors?

Do you see the same issue if you run `npm run build` locally? (delete `.next` before running that command)

**Reply by arooj-12** [reply]
yes i deleted .next folder, even package-locke.json and node modules, nothing is working. my project is ecommerce-admin(backend) which i deployed successfully on vercel and the env file contains the url of clerk auth, frontend url, cloudinary and database_url which is coming from planetscale prisma. but when i deploy ecommerce-store(frontend) i get error of prerender and its env file only contains the backend url (which i edited to vercel one). this is the error log that i am getting on vercel: Nov 27 05:23:21 PM  Error occurred prerendering page "/_not-found". Read more: https://nextjs.org/docs/messages/prerender-error
Nov 27 05:23:21 PM  SyntaxError: Unexpected token I in JSON at position 0
Nov 27 05:23:21 PM      at JSON.parse (<anonymous>)
Nov 27 05:23:21 PM      at parseJSONFromBytes (node:internal/deps/undici/undici:4553:19)
Nov 27 05:23:21 PM      at successSteps (node:internal/deps/undici/undici:4527:27)
Nov 27 05:23:21 PM      at fullyReadBody (node:internal/deps/undici/undici:1307:9)
Nov 27 05:23:21 PM      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
Nov 27 05:23:21 PM      at async specConsumeBody (node:internal/deps/undici/undici:4536:7)
Nov 27 05:23:21 PM      at async S (/opt/render/project/src/.next/server/chunks/647.js:15:4100)
Nov 27 05:23:21 PM  
 ✓ Generating static pages (6/6)
Nov 27 05:23:21 PM  
Nov 27 05:23:21 PM  > Export encountered errors on following paths:
Nov 27 05:23:21 PM      /(routes)/cart/page: /cart
Nov 27 05:23:21 PM      /_not-found
Nov 27 05:23:21 PM  ==> Build failed 😞

before deploying anything on vercel, everything was running perfect on localhost without any errors. but after i did deployment and tried running ecommerce-admin(backend) on localhost i got error related to prisma that the " Error: 
Invalid `prisma.store.findFirst()` invocation:
database provided credentials are not valid". so i changed the password on planetscale and which made me have new database_url but still i got same error on frontend and after this when i tried running again backend on localhost i got the prisma not valid error. please i need help. nobody is able to help me out on this one:////

**Reply by arooj-12** [reply]
yes i still see the same when i do npm run build locally.     TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11372:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: Error: connect ECONNREFUSED ::1:3000
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    errno: -61,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '::1',
    port: 3000
  }
}

Error occurred prerendering page "/cart". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11372:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
- info Generating static pages (4/4)

> Export encountered errors on following paths:
        /(routes)/cart/page: /cart 

i even cloned down the repo of the project i was working on but still i got the same exact error

**Reply by arooj-12** [reply]
it got resolved. i changed my planetscale password which made me have the new url for database to put on env. while my frontend and backend were running on localhost, i deployed on vercel and it worked
